database/sql/src/main/scala/cromwell/database/slick/tables/WorkflowMetadataSummaryEntryComponent.scala

```diff
@@ -134,6 +134,27 @@ trait WorkflowMetadataSummaryEntryComponent {
       if workflowMetadataSummaryEntry.workflowExecutionUuid === workflowExecutionUuid
     } yield workflowMetadataSummaryEntry.metadataArchiveStatus)
 
+  def workflowsToArchiveThatEndedOnOrBeforeThresholdTimestamp(workflowStatuses: List[Option[String]],
+                                                              workflowEndTimestampThreshold: Timestamp,
+                                                              batchSize: Long): Query[WorkflowMetadataSummaryEntries, WorkflowMetadataSummaryEntry, Seq] = {
+    (for {
+      summaryEntry <- workflowMetadataSummaryEntries
+      if workflowStatuses.map(summaryEntry.workflowStatus === _).reduce(_ || _)
```

Comment:
This does generate the right SQL:
```
select "WORKFLOW_EXECUTION_UUID", "WORKFLOW_NAME", "WORKFLOW_STATUS", "START_TIMESTAMP", "END_TIMESTAMP", "SUBMISSION_TIMESTAMP", "PARENT_WORKFLOW_EXECUTION_UUID", "ROOT_WORKFLOW_EXECUTION_UUID", "METADATA_ARCHIVE_STATUS", "WORKFLOW_METADATA_SUMMARY_ENTRY_ID" 
from "WORKFLOW_METADATA_SUMMARY_ENTRY" 
where (((("WORKFLOW_STATUS" = 'Succeeded') or ("WORKFLOW_STATUS" = 'Aborted')) or ("WORKFLOW_STATUS" = 'Failed')) 
and ("METADATA_ARCHIVE_STATUS" is null)) 
and ("END_TIMESTAMP" <= {ts '2021-04-29 14:54:25.716882'}) 
order by "WORKFLOW_METADATA_SUMMARY_ENTRY_ID" limit 1
```
I tried `contains` and it doesn't work. I think the reason is that since `workflowStatuses` is `List[Option[String]]` and `summaryEntry.workflowStatus` is `Rep[Option[String]]`, simple comparison using `==` (which `.contains` uses) doesn't work. We need `===` to check the value of the object.